src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1024,6 +930,18 @@ MasterIrreducibleExpressionWalker(Node *expression, WalkerState *state)
 }
 
 
+static bool
+MasterIrreducibleExpressionFunctionChecker(Oid func_id, void *context)
+{
+	char volatileFlag = func_volatile(func_id);
+	char *volatileContext = (char *) context;
+
+	*volatileContext = MostPermissiveVolatileFlag(*volatileContext, volatileFlag);
+
+	return (volatileFlag == PROVOLATILE_VOLATILE);
+}
```

Comment:
Hm, there's no other `MostPermissiveVolatileFlag()` callers. And this looks complicated.  I think I'd just check if `*volatileContext` is true and return, and only otherwise check if `func_volatile() == PROVOLATILE_VOLATILE``